### ISSUE TITLE
Call back for custom regularizer Coefficient

### ISSUE THREAD

**Issue by nbansal90** [keras-team/keras#8793]
I, for my application have created a custom Regularizer. And I wish to vary the "weight decay constant" after every certain number of epcohs. Is this implementation possible with callback ?

Example code:
============
from keras import backend as K

def l1_reg(weight_matrix):
    return **0.01** K.sum(K.abs(weight_matrix))

model.add(Dense(64, input_dim=64,
                kernel_regularizer=l1_reg))
model.compile()

Now I want to vary the **0.01** in the l2_reg, after every certain number epcohs.

I have tried a similar approach which was followed for decaying learning rate. But I am not able to implement using that idea, correctly. Any suggestions would be really helpful.

Thanks!

**Comment by fchollet** [comment]
As long as your updates are low-frequency (e.g. once per epoch) you could do:

```python
from keras import backend as K
import keras

l1_rate = K.variable(0.01)

def dynamic_l1(w):
   return l1_rate * K.sum(K.abs(w))

cbk = keras.callbacks.LambdaCallback(on_epoch_end=lambda epoch, logs: K.set_value(l1_rate, 0.01 * (epoch + 1)))

model.fit(..., callbacks=[cbk])
```

This would set the rate to `0.01 * (epoch + 1)` after every epoch (increase it by 0.01 after each epoch)

Warning: untested etc.

**Comment by nbansal90** [comment]
@fchollet Thank You! I have checked its working !! I have tried something on this line.

##Function to implement the decay  rate annealing
def decay_schedule(epoch):
        #Initial Learning rate 
        d_r = 0.01
        if epoch > 100:
                d_r = 0.00001 * d_r
        elif epoch > 70:
                d_r = 0.0001 * d_r
        elif epoch > 50:
                d_r = 0.001 * d_r
        elif epoch > 30:
                d_r = 0.01 * d_r

        print ('Decay rate:', d_r)
        return d_r


class DecayRate_Controller(keras.callbacks.Callback):
        def __init__(self, controller):
                super().__init__()
                self.controller = controller

        def on_epoch_begin(self, epoch, logs=None):
                d_r = self.controller(epoch)
                K.set_value(d_rate, d_r)
